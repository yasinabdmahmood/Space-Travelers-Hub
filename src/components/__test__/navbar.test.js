import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import NavBar from '../navbar/Navbar';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <NavBar />
      </Provider>
    </BrowserRouter>,

  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render the main container', () => {
  render(
    <BrowserRouter>
      <Provider store={store}>
        <NavBar />
      </Provider>
    </BrowserRouter>,
  );

  const container = screen.getByTestId('main-container');

  expect(container).toBeInTheDocument();
});
