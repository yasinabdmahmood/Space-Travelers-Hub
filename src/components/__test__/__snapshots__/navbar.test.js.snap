// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<nav
  className="navbar"
  data-testid="main-container"
>
  <div>
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <span>
      Space Travelers Hub
    </span>
  </div>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="passive"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className="passive"
        href="/profile"
        onClick={[Function]}
      >
        My profile
      </a>
    </li>
  </ul>
</nav>
`;
